### DISCUSSION TITLE
[Vercel AI-SDK] Allow setting the Generation Name

### DISCUSSION THREAD

**Discussion by JeongJuhyeon** [langfuse/langfuse#4857] [Ideas]
Is it possible to set generation names when using the Vercel AI SDK integration with Langfuse?

While it's clearly possible to set the span name through [functionId](https://langfuse.com/docs/integrations/vercel-ai-sdk#customization), span names are not visible in the Generations table UI, making it difficult to search for multiple instances of the same prompt being executed, so we can't really perform comparative analysis across similar generations.

Also see https://github.com/orgs/langfuse/discussions/4856

**Comment by hassiebp** [comment]
Hi @JeongJuhyeon - this is currently a limitation of the AI SDK integration. You cannot set the name of a generation nested in the execution trace. 

One option would be to allow passing a 'langfuseGenerationName' property in `experimentalTelemetry.metadata` to overwrite generation names. However, this would overwrite _all_ generations that occurred in a given AI SDK execution. Could you please explain what overwrite you'd like to have for your specific case for the generation names?

**Reply by JeongJuhyeon** [reply]
Hi @hassiebp , thanks for the reply.

Our usecase is pretty basic, probably the most common one. If we have a certain LLM task/pipeline consisting of multiple LLM generations, we want to have a name for each of these generations (the generation name) as well as one for the task/pipeline (the trace). To give one example: Task/trace might be "reply-to-github-issue", with two individual generation steps, "analyze-issue" and "write-reply". Currently, when using the AI SDK, for the generation steps we have to use the span name, but the problem with this is that in the Langfuse UI the span name is 1. Only exposed in one UI (the timeline), not in the tables which we mainly use 2. This also means we can't filter or search for them. Let's say we want to compare multiple generations of the "analyze-issue" step, a common need.

If this should be approached from a different angle, please let us know.

**Reply by hassiebp** [reply]
Thanks for elaborating @JeongJuhyeon - i'll convert this issue to a feature request. As an FYI, we have a generalized observation table on our roadmap that should improve the filtering here. No ETA yet on this though.

**Reply by ysuliman** [reply]
(This should be closed)
